extensions/Webhooks/lib/Webhook.pm

```diff
@@ -79,24 +79,33 @@ sub component_id {
 
 sub product {
   my ($self) = @_;
-  return $self->{product} ||=
-    Bugzilla::Product->new({id => $self->{product_id}, cache => 1});
+  return $self->{product} if exists $self->{product};
```

Comment:
Does the second return actually have an effect? My non-Perl brain sees this as returning `null` or `$self->{product}`, with the second return being unreachable. Maybe Perl just skips the line if the condition is false?

I'd lean towards making this an `if` with the return inside for readability, like:

```
if exists $self->{product}
    return $self->{product}
endif
```

If this is a standard Perl thing that I'm just not familiar with it's fine to keep. :+1:

Reply:
Yeah Perl allows for you to shorten an 'if' statement by writing it backwards. So you example you wrote is exactly the same thing. $self is essentially a hash (dictionary) and $self->{product} is a key of the hash. If the key exists, we have ran product() at least once before and can return the value early without looking up again. 

If I did `return $self->{product} if $self->{product};` then it would return early if it was a true value such as an object. But since in our case $self->{product} can also be an undefined (false) value and still be valid, then we just return it if it exists at all.